Dev > Beta

Fourdee opened the pull request:
### Changelog
#### Image line-up
- We removed Jessie images from our download server (besides for Odroid C1), since we see no advantage to use them anymore. Existing Jessie systems are still supported, but we encourage all users to consider switching to a fresh Stretch image by times: https://github.com/Fourdee/DietPi/issues/2332

#### Changes / Improvements / Optimisations
- DietPi-PREP | Removed the option to install a Jessie system. Removed the option to install DietPi on top of a Wheezy image, since an upgrade over two distro version would be required, which is not reliable. It is still possible to install DietPi on top of a Jessie image, where a distro upgrade to Stretch would be applied: https://github.com/Fourdee/DietPi/issues/2332
- General | RPi-Update: In our effort to improve system stability, and, offer software installations which build custom modules (eg: WireGuard), we can not longer support systems with non-stock APT kernel installed. If "rpi-update" is installed during update, DietPi will prompt and return you back to stock APT kernel.
- General | Moved the remaining conf/* files located on DietPi-RAMdisk to /var/lib/dietpi or into DietPi-PREP, and removed /DietPi/dietpi/conf completely. These files were only used once or never, thus have no business inside the RAMdisk.
- DietPi-Config | RPi: Improved available options for HDMI boost setting: https://github.com/Fourdee/DietPi/issues/2350
- DietPi-Config | RPi: Headless display mode is now applied via undocumented config.txt setting. Removed/Replaced some other obsolete config.txt entries: https://github.com/Fourdee/DietPi/pull/2402
- DietPi-Config | WiFi Hotspot: You can now toggle support for 802.11 N. Please note this is device and compatibility dependent, mileage may vary.
- DietPi-Drive_Manager | Now allows to check & repair the boot file system and trigger it for root file system on next reboot: https://github.com/Fourdee/DietPi/issues/1740#issuecomment-388325204
- DietPi-Software | DietPi-NordVPN: Now available for installation. Includes a GUI which allows for easy connection to a list of NordVPN servers 'dietpi-nordvpn'. If you need a NordVPN subscription, please use this link: https://go.nordvpn.net/aff_c?offer_id=15&aff_id=5305&url_id=902
- DietPi-Software | SickRage has been replaced by Medusa, which is now available for install: https://github.com/Fourdee/DietPi/issues/2239
- DietPi-Software | Blynk: Reinstalls now preserve existing config files; New directory structure and blynk user (v6.19) is patched now as well to existing installs: https://github.com/Fourdee/DietPi/pull/2324
- DietPi-Software | Plex Media Server: Updated to v1.14.0 on x86 images, switch to HTTPS ARM repo and automated locale switch to en_US.UTF-8: https://github.com/Fourdee/DietPi/issues/2294
- DietPi-Software | Nextcloud Talk: Disabled the very verbose coturn logging by default to reduce disk I/O. This can be overridden via /etc/turnserver.conf: https://github.com/Fourdee/DietPi/issues/2321
- DietPi-Software | Deluge: UMask 002 applied to all downloads: https://github.com/Fourdee/DietPi/issues/2339
- DietPi-Software | Netdata: Updated to v1.11.1 and now runs as user "netdata": https://github.com/Fourdee/DietPi/pull/2337
- DietPi-Software | ownCloud/Nextcloud: Updated webserver configs to match current recommendations and security hardenings. Only applied on new installs. To apply manually, run "dietpi-software reinstall 47" (owncloud) or "dietpi-software reinstall 114" (Nextcloud). You will be informed about the new configs, which then need to be manually moved to overwrite the old ones, since we don't want to mess with manual changes: https://github.com/Fourdee/DietPi/pull/2361
- DietPi-Software | OpenSSH: Host keys won't be recreated anymore on reinstall, only on first boot of a fresh image, using now "dpkg-reconfigure openssh-server" to generate default key types based on distro.
- DietPi-Software | Dropbear: Now installs only the "dropbear-run" package on Stretch (and above), which matches the default DietPi setup on our images and PREP script.
- DietPi-Software | NAA Daemon: Updated to 3.5.5-39. Thanks to @volpone for the heads up: https://github.com/Fourdee/DietPi/issues/2387#issue-395321320
- DietPi-Software | MPD: Updated to 0.20.23-1. Includes support for NFS and Samba: https://github.com/Fourdee/DietPi/issues/2377
- DietPi-Software | MPD: Now runs as "mpd" user again, updated systemd unit to match official one and better handle existing cache and log files on reinstall: https://github.com/Fourdee/DietPi/issues/2378
- DietPi-Software | Samba: Moved disk cache to RAM to reduce disk I/O. Thanks to @WolfganP for suggesting this enhancement: https://github.com/Fourdee/DietPi/issues/2396
- DietPi-Software | Kodi RPi: Default GPU memory split is now 320MB for 1GB devices. This should improve overall performance and stability: https://github.com/Fourdee/DietPi/issues/2365
- DietPi-Software | Airsonic: Now installs the latest upstream version automatically: https://github.com/Fourdee/DietPi/pull/2410
- DietPi-Software | You will now receive a whiptail prompt, if config files are overwritten during software installations: https://github.com/Fourdee/DietPi/issues/2325
- DietPi-Software | NoMachine: Enabled ARMv8 support and will now install current version 6.4.6: https://github.com/Fourdee/DietPi/pull/2408
- DietPi-Update | Added optional ability to automatically update DietPi, when updates are available. Please note, this is disabled by default.
- DietPi-Sync | Removed the compression option, which has no effect on local sync, since files are not stored compressed, but only transferred compressed through remote sync protocols, which are currently not offered by DietPi-Sync.

#### Bug Fixes
- DietPi-Software | Nextcloud: Resolved an issue, where the Nextcloud Apache config was not downloaded correctly. Thanks to @Stefan3v for reporting this issue: https://github.com/Fourdee/DietPi/issues/2383
- DietPi-Software | ownCloud/Nextcloud (Talk): Resolved an issue, where occ/ncc commands could fail, if Redis server is not running: https://github.com/Fourdee/DietPi/issues/2321
- DietPi-Software | ownCloud/Nextcloud: Resolved an issue, where during fresh installs on v6.19, Apache configs were not enabled correctly.
- DietPi-Software | MPD: Resolved an issue on Stretch, where the mpd binary could not find the configuration file without giving it explicitely as argument. Thanks to @mfeif for reporting this issue: https://github.com/Fourdee/DietPi/issues/2378
- DietPi-Software | Mopidy: Resolved an issue, where playlist files could not be created due to missing permissions. Further improved handling of existing configs, data and cache directories on (re)install. Thanks to @cyberlussi for reporting this issue: https://github.com/Fourdee/DietPi/issues/2384
- DietPi-Sync | Resolved an issue, where dry-run and compression options were not applied correctly. Thanks to @WilburWalsh for reporting the issue and identifying the faulty code: https://dietpi.com/phpbb/viewtopic.php?f=11&t=5347
- DietPi-Sync | Resolved an issue, where daily sync was not applied due to changed settings file scheme.
- General | Enhanced and fixed some issue with dependencies and boot order of DietPi systemd units: https://github.com/Fourdee/DietPi/pull/2357
- General | AudioPhonics I-Sabre-K2M: Once the driver is compiled, it will no longer be recompiled. This prevents ALSA installs from re-compiling the driver each time.
- General | Resolved an issue, where login with non-bash session caused an error message. Thanks to @Elijahg for reporting this issue: https://github.com/Fourdee/DietPi/issues/2347
- General | Resolved an issue where G_CHECK_URL would not terminate applications on failure. Also added option to retry + change networking options.

As always, many smaller code performance and stability improvements, visual and spelling fixes have been done, too much to list all of them here. Check out all code changes of this release on GitHub: https://github.com/Fourdee/DietPi/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+base%3Amaster

#### Known/Outstanding Issues
- DietPi-Config | RPi | OpenGL/VC4 causes sound card selection to be invalid: https://github.com/Fourdee/DietPi/issues/2173
- Pine A64 | No sound over 3.5mm jack: https://github.com/Fourdee/DietPi/issues/2153
- Odroid C2 | Some WiFi adapters do no work as hotspot: https://github.com/Fourdee/DietPi/issues/1955
- DietPi-Software | Node-RED: Pre-installed modules cannot be updated via webUI: https://github.com/Fourdee/DietPi/issues/2073
- DietPi-Software | Raspimjpeg does not stream mjpeg with Lighttpd: https://github.com/Fourdee/DietPi/issues/1747
- For all additional issues that may appear after release, please see the following link and select the 'Next Version Milestone' to see active tickets: https://github.com/Fourdee/DietPi/milestones

Commit:
v6.20

+ Tweaks

Commit:
v6,20

+ Minor

Commit:
v6.20

+ Testing, not finished.

Commit:
v6.20

+ Patch fixes. Does not re-run current patch unless this is set.

Commit:
v6.20

+ Correction.

Commit:
v6.20

+ Function code for additional uses in future.

Commit:
v6.20

+ WIP/testing

Commit:
v6.20

+ Outside func.

Commit:
v6.20

+ WIP/testing

Commit:
v6.20

+ WIP/testing

MichaIng committed:
v6.20

+ DietPi-Software | Netdata: Fix uninstall removing root dir
+ DietPi-Software | Netdata: Use netdata user to run service
+ Netdata 1.11.1 binaries.
+ CHANGELOG | Netdata update
+ DietPi-Patch | Reinstall Netdata to apply new user, permissions and service
+ DietPi-Patch | Cleanup doubled/obsolete reinstalls
+ DietPi-Software | Netdata: Simplify binary download based on HW arch; Apply permissions according to docs and logs
+ DietPi-Software | Netdata: Adjust to match official unit + hardenings from Debian sid repo unit
+ DietPi-Software | Netdata: Workaround Jessie not supporting some systemd unit directives

MichaIng committed:
v6.20

+ DietPi-FS_partition_resize | Run unit before local-fs.target and swap targets reached, but after local-fs-pre.target to have all initial R/O root mounts, systemd-fsck and remounts done
+ DietPi-FS_partition_resize | Pull in local-fs-pre.target, which is special static systemd target, that is only reached, if actively pulled in. On fresh DietPi it it reached already, but we need to be failsafe
+ DietPi-FS_partition_resize | "-f" short for "--force" forces changes, even if drive is mounted/swap active, but "--no-reread" will simply skip the test, where we know the outcome on RootFS. To be fail-safe we leave both options inside.
+ DietPi-RAMdisk/log | Simplify: "After=local-fs.target" is added via default dependencies, and all mounts have to finish to reach this target; "StandardOutput=tty" does not make sense, if we forward everything to log file
+ DietPi-RAMdisk/log | Fix "dietpi-ramdisk/log.log" being overwritten on boot, rendering shutdown log useless. For now stay with adding logs to final log location directly.
+ DietPi-FS_partition_resize | Pre-create log dir to be fail-safe
+ DietPi-(Pre)Boot | Log STDERR to log file as well
+ DietPi-Boot | Run "Before=getty", and with this on idle (can happen early by chance) login prompts
+ DietPi-Boot | Do not run "After=network-online.target", since this can delay boot much and DietPi-Boot has it's own optional (dietpi.txt) network wait mechanism. Furthermore "network-online.target" is not pulled in by default, but an "active" unit, thus consumer services need/should pull it via Wants=. "apt-daily.service" e.g. does this, but we mask this one.
+ DietPi-PreBoot | "network-pre.target" is a passive unit that needs to be pulled in (Wants=) by services, otherwise it is not active. Currently for this reason, network interfaces are raised, before/while dietpi-ramdisk/log/preboot.
+ DietPi-PostBoot | Remove "Before=rc[.-]local.service", since those are meant to run before login prompt (Before=getty.target). rc-local is enabled, login prompt will run after postboot, thus over-prompt our banner.
+ DietPi-PostBoot | Oder "After=multi-user.target", which is the last actual login prompt target and assures, that out banner is shown after login prompt TTY clearing.
+ DietPi-PostBoot/Kill_SSH | Avoid /bin/bash sub shell call, which is not required, since we do no stream redirection here
+ DietPi-Kill_SSH | "Before/After=network.target" has no order effect for this service, which is called only on shutdown, since this target is not "reached" on shutdown. "shutdown.target" is the very last target reached on shutdown, so too late anyway. Best results (earliest execution) during my tests was without declaring any ordering.
+ DietPi-Patch | Resolve networking service order cycle on Jessie
+ DietPi-Kill_SSH | Run killall regardless or pgrep result, since pgrep is slower than an unsuccessful killall attempt. Script file not required anymore. "-q" to hide output. Rename to "dietpi-kill_ssh.service" for DietPi script naming consistency.
+ DietPi-Patch | Patch DietPi-Kill_SSH service rework
+ DietPi-Kill_SSH | Do not throw failure state, if killall command fails
+ CHANGELOG | DietPi systemd unit boot order/dependencies
+ DietPi-WiFi-Monitor | Shorten "Description" to have better syslog entries; Move explanation to comment
+ DietPi-WiFi-Monitor | Since this systemd unit prints to syslog/journald, reduce output to connection loss + reconnect only. Print connection checks only in debug mode.
+ DietPi-Drive_Manager | Add fsck flag to /boot as well
+ DietPi-Drive_Manager | Remove nofail and x-systemd.automount from boot as well
+ DietPi-Patch | Remove "nofail" and "x-systemd.automount" options from fstab on update

MichaIng committed:
v6.20

+ config.txt | Be more specific about RAMdisk hint, this is does not apply to all files in /boot/
+ config.txt | Minor wording and visual

MichaIng committed:
v6.20

+ dietpi.txt | Be more specific about RAMdisk hint, this is does not apply to all files in /boot/

MichaIng committed:
v6.20

+ boot_c2.ini | Be more specific about RAMdisk hint, this is does not apply to all files in /boot/

MichaIng committed:
v6.20

+ boot_c1.ini | Add hint to edit files at RAMdisk location

MichaIng committed:
v6.20

+ boot_xu4.ini | Add hint to edit files at RAMdisk location

Commit:
v6.20

+ WIP/testing

Commit:
v6.20

+ Generate GPL license in PREP as file.

Commit:
v6.20

- General | Resolved an issue where G_CHECK_URL would not terminate applications on failure. Also added option to retry + change networking options.

Commit:
v6.20

+ Minor tweak/cleanup after use.

Commit:
v6.20

+ REmove G_CHECK_NTP
+ Improve run_ntpd code to handle blocking retry until sucess.

Commit:
v6.20

+ Minor.

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Security hardenings based on official Apache and Nginx configs
+ DietPi-Software | Nextcloud: Remove "Referrer-Policy" header, which is set internally since NC15 and not checked on NC13 (Jessie)

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Update Nginx config to match current official docs
+ DietPi-Software | Nextcloud: Remove "preload" from HSTS header. Users should decide this by themselves, since removal from preload list can take a long time, if not wanted. Further only enable HSTS header, if chosen via dietpi-letsencrypt.

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Update Nginx config to match current official docs
+ DietPi-Software | ownCloud: Remove "preload" from HSTS header. Users should decide this by themselves, since removal from preload list can take a long time, if not wanted. Further only enable HSTS header, if chosen via dietpi-letsencrypt.

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Minor Lighttpd config fix

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Lighttpd config for security hardening

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Add link to docs with Apache config

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Add link to docs with Apache config

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Add new Lighttpd config install
+ DietPi-Software | ownCloud/Nextcloud: Fix enabling old Apache site; Enable required Lighttpd modules; Update doc links
+ DietPi-Software | ownCloud/Nextcloud: Minor coding and wording + use shortest possible forum docs links with post ID + jump label only

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud/Nextcloud: Enable HSTS header in Nginx configs, if dietpi-letsencrypt HSTS setting was chosen
+ DietPi-Software | ownCloud/Nextcloud: Enable HTTPS param (strictly requires HTTPS then) only, if redirection was chosen
+ DietPi-Software | Nextcloud: Run "ncc db:convert-filecache-bigint" which is required manually one time after fresh Nextcloud install

MichaIng committed:
v6.20

+ DietPi-LetsEncrypt | Enable related ownCloud/Nextcloud Nginx settings, if HTTPS redirect resp. HSTS was chosen
+ DietPi-LetsEncrypt | Do not add preload flag to HSTS header. This should be actively decided by user, since once added, removing a domain from the list can take a long time.
+ DietPi-LetsEncrypt | Minor coding

MichaIng committed:
v6.20

+ DietPi-Patch | Re-enable ownCloud/Nextcloud Apache configs

MichaIng committed:
v6.20

+ CHANGELOG | ownCloud/Nextcloud config and install updates

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Run table conversion non-interactive

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Disable on Buster, since PHP7.3 is currently not yet supported

MichaIng committed:
v6.20

+ DietPi-Services | Add PHP7.3 for Buster and PHP7.1 to be more flexible on custom upgrades

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Apply correct "fastcgi_param HTTPS" automatically based on $https variable

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Apply correct "fastcgi_param HTTPS" automatically based on $https variable

MichaIng committed:
v6.20

+ DietPi-LetsEncrypt | HTTPS fastcgi_param is now applied automatically based on Nginx $https variable

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud/Nextcloud: fastcgi_param HTTPS is now sed automatically correctly based on Nginx $https variable

MichaIng committed:
v6.20

+ DietPi-Software | Nginx: Include /etc/nginx/modules-enabled/*.conf; to load dynamic Nginx modules, available since Stretch. On Jessie, however the directive does not hurt.

Fourdee committed:
Merge pull request #2361 from Fourdee/ownextcloud

DietPi-Software | ownCloud/Nextcloud: Config updates

Commit:
v6.20

+ Minor

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Minor, quicker for non-jessie to check 1 if statement, instead of 2.

Commit:
v6.20

+ Minor

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud: Simplify Nginx regex patterns, which also matches current docs

MichaIng committed:
v6.20

+ DietPi-Software | Make unrar an own install ID
+ DietPi-Software | Add medusa as replacement for SickRage

MichaIng committed:
v6.20

+ DietPi-Set_Software | Apply Medusa permissions

MichaIng committed:
v6.20

+ DietPi-Software | Medusa: Apply "dietpi" group to service

MichaIng committed:
v6.20

+ DietPi-Services | Control Meduse service

MichaIng committed:
v6.20

+ DietPi-Process_tool | Add Medusa process control
+ DietPi-Process_tool | Minor coding

MichaIng committed:
v6.20

+ DietPi-Patch | Remove SickRage patches and clear install ID state for Medusa

MichaIng committed:
v6.20

+ CHANGELOG | Medusa ready for install

MichaIng committed:
v6.20

+ DietPi-Process_tool | Max nice is 19

Commit:
v6.20

+ Minor

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng committed:
v6.20

+ DietPi-Software | Medusa: Do not set permissions that are set by "dietpi-set_software setpermissions" anyway. Remove executable flag to systemd unit instead, use default 644.

MichaIng committed:
v6.20

+ DietPi-Software | Tiny blank

MichaIng committed:
v6.20

+ DietPi-Software | Medusa: Skip libssl-dev install
+ DietPi-Software | rTorrent: Install latest ruTorrent from GitHub automatically

MichaIng committed:
v6.20

+ DietPi-Software | qBitTorrent: Use default "en" language instead of "en_GB", where several strings and WebUI elements were missing

MichaIng committed:
v6.20 (#2366)

+ DietPi-Sync | Resolved an issue with wrong created rsync_options string
+ DietPi-Cron.daily | Fix daily dietpi-sync based on new settings file scheme
+ DietPi-Sync | Resolved doubled dry run due to dry run required for space check
+ DietPi-Sync | Removed compression option, since this is only relevant for remote transfer, which we do not support yet. This might have led to wrong end user assumptions and confusion.
+ DietPi-Sync | Only run real sync, if prior dry run was successful
+ DietPi-Sync | Show now last log entry instead of last sync date only in main menu
+ DietPi-Sync | Simplify logging: Log to $FP_TARGET only, create tmp log first, move to $FP_TARGET afterwards; Return after insufficient space, to skip obsolete final whip and log file view.
+ DietPi-Sync | Be more verbose when target dir pre-creation fails, printing and logging "mkdir" error
+ DietPi-Sync | Exclude specific files/dirs only, if they are actually inside the source dir, and, correctly via relative path. Update info on exclude/include user file
+ DietPi-Sync | Do not allow to choose root "/" as source or target dir, point user towards DietPi-Backup. As well check for correct full path entry with leading slash.
+ DietPi-Sync | Do not check for existing dietpi_userdata which is not necessarily required
+ DietPi-Sync | Return to main menu, if directory selection was successful
+ DietPi-Sync | Removed separate "delete mode" and "cron choice" menu IDs. Those are too simple to require an own ID and we always want to go back to main menu afterwards.
+ DietPi-Sync | Turn $SYNC_DRY_RUN and samba related variables into local ones
+ DietPi-Sync | Minor coding and wording consistency
+ CHANGELOG | Entries

Commit:
v6.20

+ Minor

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Minor coding consistency.

Fourdee committed:
Merge branch 'dev' into medusa

MichaIng committed:
v6.20

+ DietPi-Process_tool | Re-add SickRage process handling to support existing installs

MichaIng committed:
v6.20

+ DietPi-Services | Re-add SickRage service handling keep supporting existing installs

MichaIng committed:
v6.20

+ DietPi-Set_Software| setpermissions: Re-add SickRage permissions to support existing installs

MichaIng committed:
v6.20

+ DietPi-Patch | Inform user about SickRage being replaced by Medusa

Commit:
v6.20

+ Fix netdata remote access: https://github.com/Fourdee/DietPi/issues/2336#issuecomment-450196178

MichaIng committed:
META | Add FeatHub page to README.md

MichaIng committed:
META | README: Add Medusa GitHub link

Fourdee committed:
Merge pull request #2373 from Fourdee/FeatHub

META | Add FeatHub page to README.md

MichaIng committed:
META | Spelling; Remove Grasshopper link

MichaIng committed:
META | Minor update

MichaIng committed:
META | README.md: Consistent list item marker

MichaIng committed:
META | CONTRIBUTING.md: Consistent list item marker

MichaIng committed:
META | Tiny

MichaIng committed:
META | bug_report.md: Consistancy

MichaIng committed:
META | feature_request.md: Consistancy

MichaIng committed:
META | image_request.md: Consistancy

MichaIng committed:
META | software_request.md: Consistancy

MichaIng committed:
v6.20

+ DietPi-Patch | Fix Medusa install detection when clearing it's install state

MichaIng committed:
v6.20

+ DietPi-Software | Syntax

MichaIng committed:
v6.20

+ DietPi-Software | Minor: White space + forgotten Download_Install() case

MichaIng committed:
v6.20

- DietPi-PREP | Use command assembly instead of subshell within crontab

MichaIng committed:
v6.20

- DietPi-Cron | Use command assembly instead of subshell within crontab
- DietPi-Cron | Always use double quotes around G_CHECK_VALIDINT() input $1 to avoid word splitting leading to unintended values for $2/3 min/max values
- DietPi-Cron | Minor coding

MichaIng committed:
v6.20

+ DietPi-Software | MPD: "apt-mark unhold" not required for "dpkg -i", which will even remove the "hold" tag
+ DietPi-Software | MPD: Re-use "mpd" user, create or modify it according to our needs
+ DietPi-Software | MPD: Preserve existing log and cache files; Move from /var/lib/mpd, if those are not yet symlinks, 
+ DietPi-Software | MPD: Update systemd unit to match current upstream/Debian package; Remove unsupported settings on Jessie

MichaIng committed:
v6.20

- DietPi-Software | MPD: Update "mpd.conf" to use "mpd" user

MichaIng committed:
v6.20

+ DietPi-Set_Software | MPD: Update permissions to use "mpd" user

Fourdee committed:
Merge pull request #2369 from Fourdee/medusa

DietPi-Software | Add Medusa as replacement for SickRage

Commit:
v6.20

+ Syntax

MichaIng committed:
v6.20

+ DietPi-Software | Nextcloud: Apply new config file download to forgotten Apache config

MichaIng committed:
v6.20

+ CHANGELOG | Resolved missing Nextcloud Apache config download

Commit:
v6.20

- DietPi-Software | NAA Daemon: Updated to 3.5.5-39. Thanks to @volpone for the heads up: https://github.com/Fourdee/DietPi/issues/2387#issue-395321320

MichaIng committed:
v6.20

+ DietPi-PreBoot | SSH servers: Regenerate distro/version default host keys via "dpkg-reconfigure"
+ DietPi-PreBoot | Start systemd unit before SSH servers, to have fresh SSH host keys for first session; It makes sense anyway to not start SSH servers before even "network-pre.target" has reached. Initramfs integration is another thing outside of systemd.
+ DietPi-Software | Dropbear: On Stretch, only install "dropbear-run", to match PREP as well
+ DietPi-Software | Dropbear: Generate ECDSA host key on Jessie, if not yet done before, since it is not done by default; Remove "/etc/dropbear" on uninstall, since with manually created host key, it is not removed by APT
+ DietPi-Software | OpenSSH: Do not recreate host keys on install, preserve existing ones
+ DietPi-Software | SSH servers: Enable services, just to be failsafe in case it got disabled at some point
+ DietPi-Software | SSH servers: Update SSH choice index, set uninstall marks and stop active SSH services during install step, to assure this is done as well when using: "dietpi-software install 104/105"
+ DietPi-Software | SSH servers: Always check for installed SSH servers via "dpkg", to assure uninstall is done as well for installs outside of dietpi-software
+ CHANGELOG | SSH servers: Host key generation and install rework
+ DietPi-Software | Init "local software_id=-1" to prevent potential issues since "0" is reserved for OpenSSH client

Commit:
v6.20

- DietPi-Software | MPD: Updated to 0.20.23-1. Includes support for NFS and Samba: https://github.com/Fourdee/DietPi/issues/2377

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng committed:
v6.20

+ DietPi-Patch | Avoid doubled dietpi-update restart when updating from pre-v6.17
+ DietPi-Patch | Remove doubled MPD + NAA reinstall
+ DietPi-Patch | Recover lost first NetData reinstall comment; Leave those on place for reference
+ DietPi-Patch | Add link to reinstall table to allow quick removal of older/obsolete reinstall

MichaIng committed:
v6.20

+ DietPi-Patch | Adjust "rc-local.service" to match current order and style and remove DietPi from requirements: https://github.com/Fourdee/DietPi/pull/2357

Commit:
v6.20

+ additional pre-reqs: https://github.com/Fourdee/DietPi/issues/2377

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Fourdee committed:
Merge branch 'dev' into mpd

MichaIng committed:
v6.20

+ DietPi-Software | MPD: Simplify download links based on $G_HW_ARCH_DESCRIPTION and resolve Jessie + ARMv8 symlink
+ DietPi-Software | MPD: Move "mpd" user home to real new cache dir in dietpi_userdata to place potential .config/.cache/.something data from 3rd party libs

MichaIng committed:
v6.20

+ DietPi-Software | Mopidy: Move our "mopidy.conf" into separately downloadable file
+ DietPi-Software | Mopidy: Change playlists dir to `/mnt/dietpi_userdata/Music`, since this must be writeable for `mopidy` user, which `/mnt` is not.
+ DietPi-Software | Mopidy: Only set config values that we do not want to match the defaults, leave all others untouched
+ DietPi-Software | Mopidy: Install our config file only if non-existent, prior to APT install, to preserve user custom config; Only assure cache and data dir within new users home dir within dietpi_userdata
  - `~/.config/mopidy/mopidy.conf` is not required. It is only used when running the `mopidy` command as root user. Since we run it as `mopidy` user and `mopidyctl`, we don't require any other user-specific configs.
+ DietPi-Software | Mopidy: Correctly add user to "dietpi" group; "usermod" needs to be used, since APT install will already create the "mopidy" user
+ DietPi-Software | Mopidy: Add "dietpi" group to systemd unit via drop-in, while leaving the default APT installed systemd unit untouched
+ DietPi-Software | Mopidy: Do not pre-create/chown cache dir on service start, since is located in non-tmp location
+ DietPi-Software | Mopidy: Move obsolete "mopidy" users home and data+cache dirs to new location, if those do not yet exist
+ DietPi-Set_Software | setpermissions: Set Mopidy home/data+cache dir to root group, since no other software must access them (and "mopidy" group is not created)
+ CHANGELOG | Mopidy playlist permission fix and install enhancement

MichaIng committed:
v6.20

+ CHANGELOG | Run MPD as "mpd" user and further installer updates

MichaIng committed:
v6.20

+ DietPi-Software | MPD: Whoopsie, add forgotten ".deb" ending

MichaIng committed:
v6.20

+ DietPi-Software | MPD: Do not try to move non-existent files/from non-existent dirs; Do not overwrite existing files

MichaIng committed:
v6.20

+ DietPi-Software | YMPD: Install Stretch binary on Buster, until we have build a Buster binary
+ DietPi-Software | YMPD: Install required libssl1.1 from Stretch repo on Jessie
+ DietPi-Software | YMPD/Shairport-Sync: On Jessie+ARMv6, install armhf libssl1.1 binary, TESTING required!
+ DietPi-Software | YMPD/myMPD: Within systemd units on Buster, comments in same line after User= is not allowed
+ DietPi-Software | myMPD: useradd "mympd" if non-existent, otherwise usermod according to our needs
+ DietPi-Software | Remove User=root entries from all systemd units, which is default anyway

MichaIng committed:
v6.20

+ DietPi-Software | YMPD: Create "ympd" user only, if non-existent, otherwise use "usermod" to assure group "dietpi" and "nologin" shell

MichaIng committed:
v6.20

+ DietPi-Software | MPD: On Stretch (custom build), symlink /etc/mpd.conf to /usr/local/etc/mpd.conf, where /usr/local/bin/mpd by default searches

MichaIng committed:
v6.20

+ CHANGELOG | Resolved MPD config file issue on Stretch

MichaIng committed:
Merge branch 'dev' into mpd

MichaIng committed:
v6.20

+ DietPi-Software | MPD: Remove config files on uninstall; Especially required on Stretch, where due to custom build non is created/removed with the deb package

Fourdee committed:
Merge pull request #2381 from Fourdee/mpd

DietPi-Software | MPD: Update installer

Commit:
v6.20

- DietPi-Config | WiFi Hotspot: You can now toggle support for 802.11 N. Please note this is device and compatibility dependent, milage may vary.

+ minor clean up

Commit:
v6.20

+ Minor visual

Commit:
v6.20

+ Minor + visual

MichaIng committed:
v6.20

+ DietPi-Software | WiFi Hotspot: Avoid doubled hostapd.conf entries

MichaIng committed:
v6.20

+ DietPi-Config | Minor coding

Commit:
v6.20

+ Minor code optimize.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng committed:
v6.20

+ DietPi-PREP | Moved htop conf into PREP and install our defaults to "/etc/htoprc". htop will create "~/.config/htop/htoprc" for every user automatically based on /etc/htoprc when opened the first time..

MichaIng committed:
v6.20

+ DietPi-PREP | Moved "htoprc" into PREP

MichaIng committed:
v6.20

+ DietPi-PREP | Moved initial "/etc/network/interfaces" into PREP. This will be re-created by DietPi-Config

MichaIng committed:
v6.20

+ DietPi-PREP | Moved initial /etc/network/interfaces into PREP

MichaIng committed:
v6.20

+ General | Move default cava.conf from RAMdisk to disk

MichaIng committed:
v6.20

+ General | Move default mpd.conf from RAMdisk to disk

MichaIng committed:
v6.20

+ DietPi-Software | MPD/Cava: Copy default config from new disk location

MichaIng committed:
v6.20

+ DietPi-Patch | Remove obsolete dietpi/conf dir from RAMdisk and disk
+ DietPi-Patch | Minor coding: Apply bash argument duplication for removal/chmod from RAMdisk and disk location at once

MichaIng committed:
v6.20

+ DietPi-Config | Minor coding

MichaIng committed:
v6.20

+ CHANGELOG | Moved the remaining conf/* files located on DietPi-RAMdisk to /var/lib/dietpi or into DietPi-PREP

Commit:
v6.20

+ Update FP for cava.conf: https://github.com/Fourdee/DietPi/pull/2393

Commit:
v6.20

+ Minor, FP remove reference.

Commit:
v6.20

+ rm last reference.

Commit:
v6.20

+ move /var/lib/dietpi/dietpi-software/conf/mpd.conf to github as download item.

Fourdee committed:
Merge pull request #2393 from Fourdee/clear_conf

General | Clean and remove /DietPi/dietpi/conf

Commit:
v6.20

+ - General | Reduced disk write count: fake-hwclock.dat now stored in RAM. Many thanks to @WolfganP! https://github.com/Fourdee/DietPi/issues/2396

MichaIng committed:
v6.20

+ CHANGELOG | Add current known/outstanding issues; Add link about Jessie support fade-out; Tiny spelling

Commit:
v6.20

+ Fix for: https://github.com/Fourdee/DietPi/issues/2396#issuecomment-451686173

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Precreate dir

Commit:
v6.20

+ Whoopsie, use existing .data

Commit:
v6.20

+ Correction

Commit:
v6.20

+ revert: https://github.com/Fourdee/DietPi/issues/2396#issuecomment-451700817

MichaIng committed:
v6.20

+ DietPi-Software | Samba: Move disk cache to RAM

MichaIng committed:
v6.20

+ CHANGELOG | Samba: Moved disk cache to RAM to reduce disk I/O
+ CHANGELOG | Removed obsolete fake-hwclock entry

MichaIng committed:
v6.20

+ DietPi-Software | OpenVPN: Increasing default key size to 2048 bit, which better matches current security standards
+ DietPi-Software | OpenVPN: Minor coding

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: Now available for install

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: APT update after adding sid repo

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: Fix sources list location

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: Add "qrencode" install, which allows easy mobile client setup via QR code on console

MichaIng committed:
v6.20

+ DietPi-Globals | Add new global dev function G_DEV_BRANCH() to apply and update to different branch

Commit:
v6.20

- DietPi-Software | DietPi-NordVPN: Now available for installation. Includes a GUI which allows for easy connection to a list of NordVPN servers 'dietpi-nordvpn'.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ PROTOCOL option

MichaIng committed:
Merge branch 'dev' into wireguard

MichaIng committed:
v6.20

+ DietPi-Software | Tiny

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: Install kernel meta packages via G_AGI to force install of new dependencies; Check whether a reboot is required due to upgraded kernel, before starting WireGuard, otherwise inform user.

MichaIng committed:
v6.20

+ DietPi-Software | MPD: Fix Jessie + ARMv8 link

Commit:
v6.20

- DietPi-Config | Static IP: Resolved an issue where settings would not be activated until after a reboot.

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng commented on dietpi/dietpi-config:
```diff
@@ -2418,7 +2418,7 @@
 		G_DIETPI-NOTIFY 2 'Restarting network, please wait...'
 		systemctl daemon-reload
-		#systemctl restart networking #: Does exactly the same as ifdown/up that we already do.
+		systemctl restart networking #: Required to apply static IP changes...
```
@Fourdee 
Are you sure that this is required? Since networking service does the same:
`ExecStart=/sbin/ifup -a --read-environment`
- And this does not even include eth0/wlan0 interfaces, if added via `allow-hotplug`, but only `auto` interfaces.
> -a, --all               process all interfaces marked "auto"

On Jessie hotplug interfaces are included (init.d service):
`if ifup -a --exclude=lo $exclusions $verbose && ifup_hotplug $exclusions $verbose`
_______
However it does not hurt, failsafe.

Fourdee replied:
@MichaIng 

Yep I got it wrong, this is required to flush DHCP IP's when changing to static:
```
		# ip addr flush dev eth$ETH_DEV_INDEX &> /dev/null
		# ip addr flush dev wlan$WIFI_DEV_INDEX &> /dev/null

```

MichaIng replied:
@Fourdee 
Hmm never found an issue when switching to STATIC or changing static IP. Of course this drops active SSH session.
ifdown and ifup reapplies `/etc/network/interface` settings, I thought, including DHCP address request, since `dhclient` is triggered via `if-up.d`.

Where did an interface restart not apply new IP?

Fourdee replied:
@MichaIng 

ASUS TB. When connected to SSH:
- Change from DHCP to static
- IP address is still the same, SSH still connected, unless ```ip addr flush dev eth0``` was used, or, reboot, then IP address updates.

Might be device specific, eg: ASUS TB networking NIC doesn't flush IP after ```ifdown``` etc?

Commit:
v6.20

+ flush required to activate new IP's
+ Force

Commit:
v6.20

+Detect if IP changed, reboot required. Was unable to get a viable (cover all) method of detection that does script exit.

Commit:
v6.20

+ Cleanup

Commit:
v6.20

+ May as well flag for reboot on exit

Commit:
v6.20

+ Reinstall samba during patch: https://github.com/Fourdee/DietPi/issues/2396#issuecomment-451701569

Fourdee committed:
Merge pull request #2397 from Fourdee/samba-cache

DietPi-Software | Samba: Move disk cache to RAM

MichaIng committed:
v6.20

+ Typo

MichaIng committed:
v6.20

+ typo

MichaIng committed:
v6.20

+ typo

MichaIng committed:
v6.20

+ typo

MichaIng committed:
v6.20

+ typo

MichaIng committed:
v6.20

+ DietPi-Config | Firmware update: On RPi give choice and do recommend to NOT use "rpi-update", but stay with APT firmware packages  instead
+ DietPi-Config | Firmware update: Check firmware (via /lib/modules) before and after update attempt, to check if it actually changed, before setting reboot requirement
+ DietPi-Config | Minor coding

MichaIng committed:
Merge branch 'dev' into wireguard

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | Do not use "rpi-update" anymore. APT packages provide regular stable updates based on the same firmware sources. Use "raspberrypi-kernel-headers" package instead of "rpi-source" to build custom kernel modules.

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: On RPi, if "rpi-update" is detected and APT did not override with new stable firmware, inform user about possibly failing kernel module and options to resolve

MichaIng committed:
v6.20

+ DietPi-Software | WireGuard: "raspberrypi-kernel-headers" never needs to be reinstalled, even if "rpi-source" was used

MichaIng committed:
v6.20

+ DietPi-Globals | Remove G_RPI_UPDATE, which is not recommended without special reason anyway and causes issues, when correct headers are required to build kernel modules

MichaIng committed:
v6.20

+ DietPi-Config | Purge "rpi-update", if APT based firmware was chosen
+ DietPi-Config | Remove ".G_RPI_UPDATE" flag creation, which is not required anymore

MichaIng committed:
v6.20

+ DietPi-Patch | Remove obsolete rpi-update flag
+ DietPi-Patch | Patch Samba cache to RAM manually, to prevent whole "dietpi-software reinstall" with config override (even we place backup)

MichaIng committed:
v6.20

+ DietPi-Config | RPi: Disable display output via undocumented config.txt setting instead of tvservice -o
+ DietPi-Config | RPi: Estimate current console resolution via vcgencmd instead of config.txt entry
+ DietPi-Config | RPi: Allow some config.txt settings to be missing, by using default values in case
+ DietPi-Config | RPi: Replace deprecated config.txt setting "display_rotate" by new "display_hdmi_rotate"
+ DietPi-Config | RPi: Minor coding

MichaIng committed:
v6.20

+ RPi | config.txt: Replace deprecated setting "display_rotate" by new "display_hdmi_rotate"
+ RPi | config.txt: Remove obsolete settings: "disable_pvt", "avoid_pwm_pll"

MichaIng committed:
v6.20

+ DietPi-Patch | Remove/Replace obsolete RPi config.txt settings

MichaIng committed:
v6.20

+ DietPi-Config | Update forgotten "display_rotate" occurrence

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | RPi: Remove edits of obsolete config.txt setting: "avoid_pwm_pll"

MichaIng committed:
v6.20

+ DietPi-PreBoot | Remove "tvservice -o", which is not achieved via new config.txt entry

MichaIng committed:
v6.20

+ DietPi-Patch | RPi: If headless display mode was selected, apply new dietpi.txt entry

MichaIng committed:
v6.20

+ CHANGELOG | DietPi-Config | RPi: Enhance and replace some obsolete dietpi.txt settings

Commit:
v6.20

- DietPi-Software | Kodi RPi: Default GPU memory split is now 320MB for 1GB devices. This should improve overall performance and stability: https://github.com/Fourdee/DietPi/issues/2365

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

Commit:
v6.20

+ Code improve https://github.com/Fourdee/DietPi/issues/2365

Commit:
v6.20

+ Minor

Commit:
v6.20

+ Minor cleanup

MichaIng committed:
v6.20

+ DietPi-Config | Fix config.txt vs dietpi.txt muddle

MichaIng committed:
v6.20

+ DietPi-Patch | Fix config.txt vs dietpi.txt muddle

MichaIng committed:
v6.20

+ DietPi-Config | CONFIG_HDMI_OUTPUT: Add an informational comment to each adjusted setting

MichaIng committed:
v6.20

+ dietpi.txt | RPi: Update CONFIG_HDMI_OUTPUT explanation

Fourdee committed:
Merge pull request #2402 from Fourdee/rpi_config

DietPi-Config | RPi: Enhance and replace some obsolete dietpi.txt settings

Fourdee commented:
Notes to self:
- 🈯️ DietPi-NordVPN https://dietpi.com/software and forum links (https://dietpi.com/phpbb/viewtopic.php?f=8&t=5&p=15975#p15975)

Commit:
v6.20

+ RC beta up

Commit:
Merge branch 'dev' of https://github.com/Fourdee/DietPi into dev

MichaIng committed:
Merge branch 'dev' into wireguard

MichaIng committed:
v6.20

+ DietPi-Globals | G_CONFIG_INJECT: Tiny failsafe + coding

MichaIng committed:
v6.20

DietPi-Set_Hardware | Add RPi headless mode option, outsourced from dietpi-config
DietPi-Set_Hardware | Add more exit paths, if input args are not supported by current device
DietPi-Set_Hardware | Minor coding

MichaIng committed:
v6.20

+ DietPi-Config | Use new DietPi-Set_Hardware rpi-headless option to apply headless video mode

MichaIng committed:
v6.20

+ DietPi-PreBoot | Apply RPi headless mode on 1st run setup, if chosen

MichaIng committed:
v6.20

+ DietPi-Patch | Use new DietPi-Set_Hardware rpi-headless option to apply headless mode, if set

MichaIng committed:
META | DietPi-Survey_report: Add new and changed v6.20 software IDs

Commit:
v6.20

+ reboot if kernel mismatch before wireguard install
+ testing

Fourdee committed:
Merge pull request #2398 from Fourdee/wireguard

DietPi-Software | WireGuard: Now available for install

Commit:
v6.20

+ WG install tweaks: https://github.com/Fourdee/DietPi/issues/2052

Commit:
v6.20

+ Uninstall tweak: https://github.com/Fourdee/DietPi/issues/2052

Commit:
v6.20

+ Minor text

Fourdee committed:
Merge pull request #2405 from Fourdee/rpi-headless

RPi | Outsource headless mode to DietPi-Set_Hardware + minor enhancements

Commit:
v6.20

+ Remove debugging
+ text

Commit:
v6.20

+ Minor fix

Commit:
v6.20

+ Force reinstall on kernel packages.

Commit:
v6.20

+  General | RPi-Update: In our effort to improve system stability, and, offer software installations which build custom modules (eg: Wireguard), we can not longer support systems with non-stock-APT kernel installed.

Commit:
v6.20

+ Minor

Commit:
v6.20

+ no longer required.

Commit:
v6.20

+ Dietpi-config remove rpi-update option
+ Patch for APT kernel install
https://github.com/Fourdee/DietPi/issues/2052

Commit:
v6.20

+ minor

Commit:
v6.20

+ https://github.com/Fourdee/DietPi/pull/2398#discussion_r247239893

Commit:
v6.20

+ Odroid's: https://github.com/Fourdee/DietPi/issues/2052

Commit:
v6.20

+ fix for https://github.com/Fourdee/DietPi/issues/2052#issuecomment-453690247

Commit:
v6.20

+ Reboot required.

EG: in case of Odroid headers do not match active kernel, thus fails to active module.

https://github.com/Fourdee/DietPi/issues/2052#issuecomment-453690247

MichaIng committed:
v6.20

+ DietPi-Config | Allow firmware update via G_AGDUG for RPi as well, so user is not forced to run dietpi-update/software or dist-upgrade manually; Otherwise menu option should be hidden on RPi.

MichaIng committed:
v6.20

+ DietPi-Patch | RPi-Update: Do no call G_AGUP again within patch, since it was already called on update start
+ DietPi-Patch | RPi-Update: Remove rpi-source as well, if found

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | i-sabre-k2m: Rebuild the kernel module each time, the sound card is selected, in case the kernel got upgraded and inform user about this requirement

MichaIng committed:
v6.20

+ DietPi-Patch | Remove obsolete .compiled_i-sabre-k2m flag as well

Fourdee committed:
Merge pull request #2407 from Fourdee/kernel_updates

DietPi-Config | Firmware update fine tuning

Commit:
v6.20

+ Provide info when config files are updated via .conf install: https://github.com/Fourdee/DietPi/issues/2325#issuecomment-453792735

Commit:
v6.20

+ go with whippy.

Commit:
v6.20

- DietPi-Software | You will now recieve a whiptail prompt, if config files are overwritten during software installations: https://github.com/Fourdee/DietPi/issues/2325

MichaIng committed:
v6.20

+ DietPi-Software | ownCloud/Nextcloud/Wordpress: Download tar.bz2/gz archives, to have preserved permissions and lower download size

MichaIng committed:
v6.20

+ DietPi-Software | NoMachine: Update installs to v6.4.6; Enable ARMv8 support, testing required!

MichaIng committed:
v6.20

+ CHANGELOG | NoMachine: Enabled ARMv8 support and will now install current version 6.4.6

MichaIng committed:
v6.20

+ DietPi-Software | Airsonic: Install latest version via api.github.com
+ DietPi-Software | Airsonic: If user already exists, update via usermod instead of attempting useradd
+ DietPi-Software | Airsonic: Move airsonic home dir to dietpi_userdata as well
+ DietPi-Software | Minor coding

MichaIng committed:
v6.20

+ CHANGELOG | Airsonic: Now installs the latest upstream version automatically

MichaIng committed:
v6.20

+ DietPi-LetsEncrypt | Add more info about supported web applications, in non was found, and allow to switch to DietPi-Software directly
+ DietPi-LetsEncrypt | Minor coding

MichaIng committed:
v6.20

+ DietPi-Software | Minor coding

MichaIng committed:
v6.20

+ DietPi-Software | Airsonic: Use wget instead of Download_Install(), since we don't download a 7z archive anymore

MichaIng committed:
v6.20

+ DietPi-Software | Airsonic: wget -O requires a destination file name

MichaIng committed:
v6.20

+ DietPi-Software | Java: "ca-certificates-java" is now dependency of "openjdk-8-jre-headless" on all distro versions, so do not manually install
+ DietPi-Software | Java: "ca-certificates-java" dependencies now allow Java 8 only on all distro versions, Java 9+10 have been dropped mostly, so we don't require reinstall workaround anymore
+ DietPi-Software | Java: Install headless packages on Jessie as well
+ DietPi-Software | Java: Add all Java related packages to backports priority pin on Jessie, to allow automated APT upgrades for all of them

MichaIng committed:
v6.20

+ DietPi-Software | Jackett: Supports /latest on api.github.com

Fourdee committed:
Merge pull request #2410 from Fourdee/airsonic

DietPi-Software | Airsonic: Update installer to latest version

Fourdee committed:
Merge pull request #2408 from Fourdee/nomachine

DietPi-Software | NoMachine: Update to v6.4.6 and enable for ARMv8
